test/Datadog.Trace.Tests/Tagging/TagsListTests.cs

```diff
@@ -5,13 +5,40 @@
 using Datadog.Trace.Agent.MessagePack;
 using Datadog.Trace.ClrProfiler.Integrations.AdoNet;
 using Datadog.Trace.Tagging;
+using Datadog.Trace.Util;
 using Datadog.Trace.Vendors.MessagePack;
 using Xunit;
 
 namespace Datadog.Trace.Tests.Tagging
 {
     public class TagsListTests
     {
+        [Fact]
+        public void SetTag_WillNotCauseDuplicates()
+        {
+            // Initialize common tags
+            var tags = new DerivedCommonTags()
+            {
+                Version = "v1.0",
+                Environment = "Test"
+            };
+
+            // Initialize custom tags
+            tags.SetTag("sample.1", "Temp 1");
+            tags.SetTag("sample.2", "Temp 2");
+
+            // Try set existing tag
+            tags.SetTag(Tags.Version, "v2.0");
+            tags.SetTag("sample.2", "Temp 3");
+
+            var all = tags.GetAllTags();
```

Comment:
Rather than adding this method, wouldn't it be better to:
 - call `GetTag` and make sure the new value of the tag is returned
 - call `SerializeTo` and check that the tag is serialized only once

In the end, we don't _really_ care if the value is stored internally once or n times (though it's still nice to confirm it), what matters the most is the value returned by the public APIs

Reply:
Yes very good point, I'll modify the test to do that

Reply:
Updated in be93545517654d92f5ccb123509b17996a9b578f . I made a slight pivot and focused this new unit test on GetTag and GetMetric. I'm leaving the serialization test to the Serialize unit test. What do you think @kevingosse ?

Reply:
LGTM 👍